File: ios/CollectionViewSample/CollectionViewSample/ViewController.m
//
//  ViewController.m
//  CollectionViewSample
//
//  Created by fujita taisuke on 2012/11/19.
//  Copyright (c) 2012年 Revolution 9. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
	// Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (NSInteger)numberOfSectionsInCollectionView:(UICollectionView *)collectionView
{
    NSLog(@"%@#%@", NSStringFromClass([self class]), NSStringFromSelector(_cmd));
    return 1;
}

- (NSInteger)collectionView:(UICollectionView *)collectionView
     numberOfItemsInSection:(NSInteger)section
{
    NSLog(@"%@#%@", NSStringFromClass([self class]), NSStringFromSelector(_cmd));
    return 12;
}

- (UICollectionViewCell *)collectionView:(UICollectionView *)collectionView
                  cellForItemAtIndexPath:(NSIndexPath *)indexPath
{
    NSLog(@"%@#%@", NSStringFromClass([self class]), NSStringFromSelector(_cmd));
    UICollectionViewCell *cell = [collectionView dequeueReusableCellWithReuseIdentifier:@"Cell"
                                                                           forIndexPath:indexPath];
    UIImage *image = [UIImage imageNamed:@"hedgehog"];
    UIImageView *view = [[UIImageView alloc] initWithImage:image];
    view.transform = CGAffineTransformMakeRotation(M_PI/180 * (30 * indexPath.row));
    [cell addSubview:view];
    return cell;
}

- (UICollectionReusableView *)collectionView:(UICollectionView *)collectionView viewForSupplementaryElementOfKind:(NSString *)kind atIndexPath:(NSIndexPath *)indexPath
{
    NSLog(@"%@#%@", NSStringFromClass([self class]), NSStringFromSelector(_cmd));
    if (![kind isEqualToString:UICollectionElementKindSectionHeader]) {
        return nil;
    }
    UICollectionReusableView *view = [collectionView dequeueReusableSupplementaryViewOfKind:UICollectionElementKindSectionHeader
                                                                        withReuseIdentifier:@"Header"
                                                                               forIndexPath:indexPath];
    UILabel *label = [[UILabel alloc] initWithFrame:CGRectMake(5, 5, 200, 40)];
    label.backgroundColor = [UIColor clearColor];
    label.text = @"画像の角度を変えてみる";
    [view addSubview:label];
    return view;
}

- (CGSize)collectionView:(UICollectionView *)collectionView layout:(UICollectionViewLayout*)collectionViewLayout sizeForItemAtIndexPath:(NSIndexPath *)indexPath {
    NSLog(@"%@#%@", NSStringFromClass([self class]), NSStringFromSelector(_cmd));
    return CGSizeMake(60, 60);
}

@end

File: ios_swift/CollectionView/CollectionView/ViewController.swift
//
//  ViewController.swift
//  CollectionView
//
//  Created by taisuke fujita on 2014/06/25.
//  Copyright (c) 2014年 taisuke fujita. All rights reserved.
//

import UIKit

class ViewController: UICollectionViewController, UICollectionViewDelegateFlowLayout {
                            
    override func viewDidLoad() {
        super.viewDidLoad()
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    override func numberOfSectionsInCollectionView(collectionView: UICollectionView) -> Int {
        return 1;
    }

    override func collectionView(collectionView: UICollectionView,
        numberOfItemsInSection section: Int) -> Int {
            return 12;
    }
    
    override func collectionView(collectionView: UICollectionView,
        cellForItemAtIndexPath indexPath: NSIndexPath) -> UICollectionViewCell {
            let cell = collectionView.dequeueReusableCellWithReuseIdentifier("Cell", forIndexPath: indexPath)
                as UICollectionViewCell
            let image = UIImage(named: "hedgehog")
            let imageView = UIImageView(image: image) as UIImageView
            let radian = CGFloat(M_PI / 180) * CGFloat(30 * indexPath.row)
            imageView.transform = CGAffineTransformMakeRotation(radian)
            cell.addSubview(imageView)
            return cell
    }
    
    override func collectionView(collectionView: UICollectionView,
        viewForSupplementaryElementOfKind kind: String,
        atIndexPath indexPath: NSIndexPath) -> UICollectionReusableView {
            if kind != UICollectionElementKindSectionHeader {
                return super.collectionView(collectionView,
                    viewForSupplementaryElementOfKind: kind,
                    atIndexPath: indexPath);
            }
            let header = collectionView.dequeueReusableSupplementaryViewOfKind(
                kind,
                withReuseIdentifier: "Header",
                forIndexPath: indexPath) as UICollectionReusableView
            let label = UILabel(frame: CGRect(x: 5, y: 5, width: 200, height: 40))
            label.backgroundColor = UIColor.clearColor()
            label.text = "画像の角度を変えてみる"
            header.addSubview(label)
            return header;
    }
    
    func collectionView(collectionView: UICollectionView!,
        layout collectionViewLayout: UICollectionViewLayout!,
        sizeForItemAtIndexPath indexPath: NSIndexPath!) -> CGSize {
            return CGSize(width: 60, height: 60)
    }
}

